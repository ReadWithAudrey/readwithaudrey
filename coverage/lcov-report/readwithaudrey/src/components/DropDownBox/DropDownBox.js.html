<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for readwithaudrey/src/components/DropDownBox/DropDownBox.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">readwithaudrey/src/components/DropDownBox</a> DropDownBox.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import PropTypes from 'prop-types'
import { Select } from './DropDownBox.style'
&nbsp;
const DropDownBox = () =&gt; {
  return (
    &lt;Select&gt;
      &lt;option value="-12"&gt;(GMT-12:00) International Date Line West&lt;/option&gt;
      &lt;option value="-11"&gt;(GMT-11:00) Midway Island, Samoa&lt;/option&gt;
      &lt;option value="-10"&gt;(GMT-10:00) Hawaii&lt;/option&gt;
      &lt;option value="-9"&gt;(GMT-09:00) Alaska&lt;/option&gt;
      &lt;option value="-8"&gt;
        (GMT-08:00) Pacific Time (US &amp; Canada), Tijuana
      &lt;/option&gt;
      &lt;option value="-7"&gt;
        (GMT-07:00) Mountain Time (US &amp; Canada),Arizona
      &lt;/option&gt;
      &lt;option value="-6"&gt;
        (GMT-06:00) Central Time, Mexico City, Central America
      &lt;/option&gt;
      &lt;option value="-5"&gt;
        (GMT-05:00) Eastern Time (US &amp; Canada), Lima, Bogota, Quito
      &lt;/option&gt;
      &lt;option value="-4"&gt;
        (GMT-04:00) Atlantic Time (Canada), Caracus, Santiago
      &lt;/option&gt;
      &lt;option value="-3.5"&gt;(GMT-03:30) Newfoundland&lt;/option&gt;
      &lt;option value="-3"&gt;
        (GMT-03:00) Buenos Aires, Montevideo, Greenland, Brasilia
      &lt;/option&gt;
      &lt;option value="-2"&gt;(GMT-02:00) Mid-Atlantic&lt;/option&gt;
      &lt;option value="-1"&gt;(GMT-01:00) Azores, Cape Verde Is.&lt;/option&gt;
      &lt;option value="0" selected="selected"&gt;
        (GMT+00:00) Greenwich Mean Time : Dublin, Edinburgh, Lisbon, London
      &lt;/option&gt;
      &lt;option value="1"&gt;
        (GMT+01:00) Amsterdam, Paris, Brussels, Rome, Berlin, Madrid
      &lt;/option&gt;
      &lt;option value="2"&gt;(GMT+02:00) Amman&lt;/option&gt;
      &lt;option value="2"&gt;
        (GMT+02:00) Athens, Amman, Beirut, Bucharest, Istanbul
      &lt;/option&gt;
      &lt;option value="3"&gt;(GMT+03:00) Kuwait, Riyadh, Baghdad&lt;/option&gt;
      &lt;option value="3"&gt;
        (GMT+03:00) Moscow, St. Petersburg, Kuwait, Nairobi
      &lt;/option&gt;
      &lt;option value="3.5"&gt;(GMT+03:30) Tehran&lt;/option&gt;
      &lt;option value="4"&gt;(GMT+04:00) Abu Dhabi, Muscat, Baku&lt;/option&gt;
      &lt;option value="4.5"&gt;(GMT+04:30) Kabul&lt;/option&gt;
      &lt;option value="5"&gt;(GMT+05:00) Islamabad, Karachi, Tashkent&lt;/option&gt;
      &lt;option value="5.5"&gt;
        (GMT+05:30) Chennai, Kolkata, Mumbai, New Delhi
      &lt;/option&gt;
      &lt;option value="5.75"&gt;(GMT+05:45) Kathmandu&lt;/option&gt;
      &lt;option value="6"&gt;(GMT+06:00) Almaty, Astana, Novosibirsk&lt;/option&gt;
      &lt;option value="6.5"&gt;(GMT+06:30) Yangon (Rangoon)&lt;/option&gt;
      &lt;option value="7"&gt;(GMT+07:00) Bangkok, Hanoi, Jakarta&lt;/option&gt;
      &lt;option value="8"&gt;
        (GMT+08:00) Beijing, Kuala Lumpur, Hong Kong, Singapore
      &lt;/option&gt;
      &lt;option value="9"&gt;(GMT+09:00) Tokyo, Seoul, Osaka&lt;/option&gt;
      &lt;option value="9.5"&gt;(GMT+09:30) Adelaide, Darwin&lt;/option&gt;
      &lt;option value="10"&gt;
        (GMT+10:00) Brisbane, Canberra, Melbourne, Sydney
      &lt;/option&gt;
      &lt;option value="11"&gt;
        (GMT+11:00) Magadan, Solomon Is., New Caledonia
      &lt;/option&gt;
      &lt;option value="12"&gt;(GMT+12:00) Auckland, Wellington, Fiji&lt;/option&gt;
      &lt;option value="13"&gt;(GMT+13:00) Nukualofa&lt;/option&gt;
    &lt;/Select&gt;
  )
}
&nbsp;
DropDownBox.propTypes = {
  children: PropTypes.string,
}
&nbsp;
export default DropDownBox
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 12 2018 12:51:26 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
