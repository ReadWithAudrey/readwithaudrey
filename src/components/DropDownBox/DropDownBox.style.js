import styled from 'styled-components'

export const StyledRadioButton = styled.button`
  background: #4ce0b3;
  border: 2px solid white;
  border-radius: 4px;
  color: white;
`

export const Select = styled.select`
  background: #fff;
  border: #d3d3d3 1px solid;
  padding: 7px;
  margin: 10px 0px;
  border-radius: 10px;
`
