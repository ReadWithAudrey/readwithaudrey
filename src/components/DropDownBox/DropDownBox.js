import React from 'react'
import PropTypes from 'prop-types'
import { Select } from './DropDownBox.style'

const DropDownBox = props => {
  if (props.type === 'timezone') {
    return (
      <Select onChange={props.onChange} name={props.name}>
        <option value="-12">(GMT-12:00) International Date Line West</option>
        <option value="-11">(GMT-11:00) Midway Island, Samoa</option>
        <option value="-10">(GMT-10:00) Hawaii</option>
        <option value="-9">(GMT-09:00) Alaska</option>
        <option value="-8">
          (GMT-08:00) Pacific Time (US & Canada), Tijuana
        </option>
        <option value="-7">
          (GMT-07:00) Mountain Time (US & Canada), Arizona
        </option>
        <option value="-6">
          (GMT-06:00) Central Time, Mexico City, Central America
        </option>
        <option value="-5">
          (GMT-05:00) Eastern Time (US & Canada), Lima, Bogota, Quito
        </option>
        <option value="-4">
          (GMT-04:00) Atlantic Time (Canada), Caracus, Santiago
        </option>
        <option value="-3.5">(GMT-03:30) Newfoundland</option>
        <option value="-3">
          (GMT-03:00) Buenos Aires, Montevideo, Greenland, Brasilia
        </option>
        <option value="-2">(GMT-02:00) Mid-Atlantic</option>
        <option value="-1">(GMT-01:00) Azores, Cape Verde Is.</option>
        <option value="0" selected="selected">
          (GMT+00:00) Greenwich Mean Time : Dublin, Edinburgh, Lisbon, London
        </option>
        <option value="1">
          (GMT+01:00) Amsterdam, Paris, Brussels, Rome, Berlin, Madrid
        </option>
        <option value="2">(GMT+02:00) Amman</option>
        <option value="2">
          (GMT+02:00) Athens, Amman, Beirut, Bucharest, Istanbul
        </option>
        <option value="3">(GMT+03:00) Kuwait, Riyadh, Baghdad</option>
        <option value="3">
          (GMT+03:00) Moscow, St. Petersburg, Kuwait, Nairobi
        </option>
        <option value="3.5">(GMT+03:30) Tehran</option>
        <option value="4">(GMT+04:00) Abu Dhabi, Muscat, Baku</option>
        <option value="4.5">(GMT+04:30) Kabul</option>
        <option value="5">(GMT+05:00) Islamabad, Karachi, Tashkent</option>
        <option value="5.5">
          (GMT+05:30) Chennai, Kolkata, Mumbai, New Delhi
        </option>
        <option value="5.75">(GMT+05:45) Kathmandu</option>
        <option value="6">(GMT+06:00) Almaty, Astana, Novosibirsk</option>
        <option value="6.5">(GMT+06:30) Yangon (Rangoon)</option>
        <option value="7">(GMT+07:00) Bangkok, Hanoi, Jakarta</option>
        <option value="8">
          (GMT+08:00) Beijing, Kuala Lumpur, Hong Kong, Singapore
        </option>
        <option value="9">(GMT+09:00) Tokyo, Seoul, Osaka</option>
        <option value="9.5">(GMT+09:30) Adelaide, Darwin</option>
        <option value="10">
          (GMT+10:00) Brisbane, Canberra, Melbourne, Sydney
        </option>
        <option value="11">
          (GMT+11:00) Magadan, Solomon Is., New Caledonia
        </option>
        <option value="12">(GMT+12:00) Auckland, Wellington, Fiji</option>
        <option value="13">(GMT+13:00) Nukualofa</option>
      </Select>
    )
  } else if (props.type === 'age') {
    return (
    <Select onChange={props.onChange} name={props.name}>
      <option value="18-24">18-24</option>
      <option value="25-34">25-34</option>
      <option value="35-44">35-44</option>
      <option value="45-54">45-54</option>
      <option value="55-64">55-64</option>
      <option value="65-74">65-74</option>
      <option value="75+">75+</option>
    </Select>
    )
  }
}

  DropDownBox.propTypes = {
    children: PropTypes.string,
    onChange: PropTypes.func,
    name: PropTypes.string,
    type: PropTypes.string,
  }

  export default DropDownBox
