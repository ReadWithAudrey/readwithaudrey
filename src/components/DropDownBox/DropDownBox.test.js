import React from 'react'
import renderer from 'react-test-renderer'
import DropDownBox from './DropDownBox'

describe('DropDownBox component snapshot test', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<DropDownBox />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
