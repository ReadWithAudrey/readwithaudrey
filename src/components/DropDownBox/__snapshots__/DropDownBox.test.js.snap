// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownBox component snapshot test renders correctly 1`] = `
<select
  className="sc-bdVaJa montserrat w-100  mv2 pa2 w-90 bJCnal"
>
  <option
    value="18-24"
  >
    18-24
  </option>
  <option
    value="25-34"
  >
    25-34
  </option>
  <option
    value="35-44"
  >
    35-44
  </option>
  <option
    value="45-54"
  >
    45-54
  </option>
  <option
    value="55-64"
  >
    55-64
  </option>
  <option
    value="65-74"
  >
    65-74
  </option>
  <option
    value="75+"
  >
    75+
  </option>
</select>
`;
